lib/rest.js

```diff
@@ -36,40 +36,64 @@ module.exports = class Rest {
             this.options.requestAttempts
         );
     }
-    isTransactionalMessageApi(url) {
+    /**
+     * helper for {@link getBulk} to determine if the url is a transactional message API
+     * @private
+     * @param {string} url
+     * @returns {boolean} true if the url is a transactional message API
+     */
+    _isTransactionalMessageApi(url) {
         return url && this.transactionalApis.some((api) => url.includes(api));
     }
     /**
      * Method that makes paginated GET API Requests using $pageSize and $page parameters
      *
      * @param {string} url of the resource to retrieve
      * @param {number} [pageSize] of the response, defaults to 50
+     * @param {string} [iteratorField] attribute of the response to iterate over ('items'|'definitions'|'entry')
      * @returns {Promise<object>} API response combined items
      */
-    async getBulk(url, pageSize) {
-        let iteratorField;
+    async getBulk(url, pageSize, iteratorField) {
         let page = 1;
         const baseUrl = url.split('?')[0];
-        const isTransactionalMessageApi = this.isTransactionalMessageApi(baseUrl);
+        const isTransactionalMessageApi = this._isTransactionalMessageApi(baseUrl);
+        const isLegacyApi = baseUrl && baseUrl.startsWith('/legacy/v1/');
```

Comment:
i would tend to prefer the same format as transactional for consistency ( _isLegacyApi)

Reply:
but you are comparing a method name with variable names here. oversight?

Reply:
just proposing a new method "_isLegacyApi" rather than the logic here.